website/docs/topics/feature-flags/best-practices-using-feature-flags-at-scale.md

```diff
@@ -0,0 +1,342 @@
+
+### 15. Do not define business logic using feature flags.
+
+In our experience, since feature flags reduce the friction of releasing software, teams who adopt feature flags tend to use them more and more.  This is a good thing as long as archived flags are removed from the code base keeping tech debt down. You should resist, however, the temptation to codify core business logic inside feature flags, the same way you should [resist wrapping application configuration in feature flags](#12-dont-use-flags-to-manage-configuration).
+
+Here is a good rule of thumb: Experiment with business logic using feature flags.  But once you have established the business logic, codify it in code that lives outside of a flag.
+
+For example, if you want to determine whether premium users of your product will use a certain new feature, you can wrap it in a flag and measure the result. Once you have determined that premium users do use the feature, and that all premium users should have access to it, however, you should remove the flag and add the feature to your entitlement service.
+
+If feature flags are so great, why shouldn’t you use them for business logic? A few reasons:
+
+#### Dependency on 3rd party services
+
+Coming from a feature flag vendor, it might surprise you that we do not advocate making core business logic dependent on a 3rd party feature flag service. If that flag service is down, then your app could potentially cease to function in the way designed.  This applies to using 3rd party feature flag services, as well as home grown feature flags service. 
+
+#### Complexity and Maintainability
+
+Embedding business logic within feature flags can make the codebase unnecessarily complex. Business rules can become scattered and entangled with feature flagging logic, making the code harder to read, understand, and maintain. When changes are needed, developers might have to navigate through a labyrinth of feature flags to find and update the relevant logic, increasing the risk of introducing bugs.
+
+#### Performance Implications
+
+Feature flags are typically designed to be checked frequently and quickly, with minimal performance overhead. However, when feature flags are used to control business logic, [they may involve more complex evaluations and data fetching, which can degrade application performance](#14-avoid-giant-feature-flag-targeting-lists). This is particularly problematic for high-traffic applications where performance is critical.
+
+#### Security Risks
+
+Business logic often involves access controls and entitlements. Using feature flags to manage these aspects can expose security vulnerabilities if not handled correctly. Feature flags might be toggled accidentally or maliciously, leading to unauthorized access or exposure of sensitive data.
+
+## Staying sane while managing feature flags at scale
```

Comment:
I'd avoid using terms related to sanity in writing (for ableism reasons). Can we rephrase this to something else? What about this?

```suggestion
## Keeping cool while managing feature flags at scale
```

This also applies in the text, by the way.